pytorch_lightning/overrides/data_parallel.py

```diff
@@ -7,6 +7,8 @@
 from torch.nn import DataParallel
 from torch.nn.parallel import DistributedDataParallel
 
+from pytorch_lightning.trainer.state import TrainerMode
```

Comment:
Nice work is done in this PR!
I think this line is buggy (well is it?), because `pytorch_lightning.trainer.state` is not visible to this module (plz refer to [ci tests](https://github.com/PyTorchLightning/pytorch-lightning/pull/770/checks?check_run_id=448544209#step:7:42)).
I have two suggestions to solve this.
The first suggestion is to make `TrainerMode` an inner class of `Trainer`.
The second suggestion is to keep your commits as they are but to make `TrainerMode` visible. That requires updating two different `__init__.py` files, the [first](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/__init__.py) is of `pytorch_lightning` and the [second](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/__init__.py) is of `pytorch_lightning.trainer`.

What do you think? @Borda @xingzhaolee @kuynzereb

Reply:
well, the problem is with a cycle in imports:
```
from .core import data_loader, LightningModule
pytorch_lightning\core\__init__.py:101: in <module>
    from .lightning import LightningModule
pytorch_lightning\core\lightning.py:17: in <module>
    from pytorch_lightning.overrides.data_parallel import LightningDistributedDataParallel
pytorch_lightning\overrides\data_parallel.py:10: in <module>
    from pytorch_lightning.trainer.state import TrainerMode
pytorch_lightning\trainer\__init__.py:28: in <module>
    from .trainer import Trainer
pytorch_lightning\trainer\trainer.py:16: in <module>
    from pytorch_lightning.trainer.distrib_parts import (
pytorch_lightning\trainer\distrib_parts.py:341: in <module>
    from pytorch_lightning.overrides.data_parallel import (
E   ImportError: cannot import name 'LightningDistributedDataParallel'
```
@xingzhaolee could you check it ^^